test/unit/BPool/BPool_JoinswapExternAmountIn.t.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from './BPoolBase.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {BNum} from 'contracts/BNum.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BPoolJoinswapExternAmountIn is BPoolBase, BNum {
+  address public tokenIn;
+
+  // Valid scenario
+  uint256 public tokenAmountIn = 1e18;
+  uint256 public tokenInWeight = 2e18;
+  uint256 public totalWeight = 10e18;
+  uint256 public tokenInBalance = 50e18;
+
+  // (((tokenAmountIn*(1-(1-tokenInWeight/totalWeight)*MIN_FEE)+tokenInBalance)/tokenInBalance)^(tokenInWeight/totalWeight))*INIT_POOL_SUPPLY - INIT_POOL_SUPPLY
```

Comment:
we could, but for that we could reference bmath directly. In these comments I explicitly implement the formulae as defined by bmath in a notation that's easily executable, using the same variable names that'll be found in the code, as both a sanity check and the way to independently compute them (this one got me searching for alternatives to`bc` tho, because of the non-integer exponent)